src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -275,15 +238,15 @@ void ProcessSelectedFields(XPathNavigator nav, TypeDefinition type)
             }
         }
 
-        protected virtual void ProcessField(TypeDefinition type, XPathNavigator nav)
+        protected virtual void ProcessField(TypeDesc type, XPathNavigator nav)
         {
             string signature = GetSignature(nav);
             if (!String.IsNullOrEmpty(signature))
             {
-                FieldDefinition? field = GetField(type, signature);
+                FieldDesc field = type.GetField(signature);
```

Comment:
`signature` is in the form of `System.Int32 field1`. TypeDesc.GetField accepts the field name. This won't work - could you keep the `GetField` method that ILLinker has and tweak it to work with the type system? Use the Cecil name formatter to format the field type name.